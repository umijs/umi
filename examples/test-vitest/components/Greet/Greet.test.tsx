import renderer from 'react-test-renderer';
import Greet from './Greet';

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

test('name prop can be rendered correctly', () => {
  const component = renderer.create(<Greet name="foo" />);
  const instance = component.root;
  expect(instance.props.name).toEqual('foo');
});

test('click event can be called correctly', () => {
  const fn = vi.fn();
  const component = renderer.create(<Greet onClick={fn} />);
  component.root.props.onClick();
  expect(fn).toBeCalled();
});

test('Greet snapshot', () => {
  const component = renderer.create(<Greet name="snapshot" />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
