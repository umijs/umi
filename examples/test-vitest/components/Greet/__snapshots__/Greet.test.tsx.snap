// Vitest Snapshot v1

exports[`Greet snapshot 1`] = `
<div>
  Hello 
  snapshot
</div>
`;
